tommath_private.h

```diff
@@ -236,8 +236,13 @@ MP_DEPRECATED(s_mp_toom_mul) mp_err mp_toom_mul(const mp_int *a, const mp_int *b
 MP_DEPRECATED(s_mp_toom_sqr) mp_err mp_toom_sqr(const mp_int *a, mp_int *b);
 MP_DEPRECATED(s_mp_reverse) void bn_reverse(unsigned char *s, int len);
 
-/* From the C-FAQ */
-#define MP_ENDIANESS ((0x41424344 == *((uint32_t*)"ABCD"))?MP_BIG_ENDIAN:MP_LITTLE_ENDIAN)
+/* Thanks to PSkocik at stackoverflow for pointing out that C++ has no compound literals */
```

Comment:
Urks. Why not use only the c++ version. I really dislike this __cplusplus macro.

Reply:
Hmm but I also don't like this static. Why not keep the ABCD version?

Reply:
Is this also not c++ compatible? This is a mess. It is not necessary for tommath to be c++ compilable. We also don't test this and compile with g++.

Reply:
Yepp, it's not the most beautiful of all the macros, so if you insist, I'll delete.

Reply:
> Why not keep the ABCD version?

Not standard compatible (alignment problems). [I asked at stackoverflow](https://stackoverflow.com/questions/58224612/standard-compliance-of-a-c-macro) to get some hints because I'm not very fluent in C++.
clang complained, too.

> It is not necessary for tommath to be c++ compilable.

If it is cheap to have&hellip;

> We also don't test this and compile with g++ in the CI.

Mmh&hellip; should we?
But now that you mentioned it, I have never tried, so
`CC=g++ make test && ./test`
A lot of warnings of the kind
```text
cc1plus: warning: command line option ‘-Wdeclaration-after-statement’ is valid for C/ObjC but not for C++
cc1plus: warning: command line option ‘-Wbad-function-cast’ is valid for C/ObjC but not for C++
cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
```
but otherwise works.

clang++-7 complains with a lot of
```text
clang: warning: treating 'c' input as 'c++' when in C++ mode, this behavior is deprecated [-Wdeprecated]
```

and throws one error

```text
   * clang++-7 demo/test.o
clang: warning: treating 'c' input as 'c++' when in C++ mode, this behavior is deprecated [-Wdeprecated]
demo/test.c:2277:22: error: invalid suffix on literal; C++11 requires a space between literal and identifier
      [-Wreserved-user-defined-literal]
   printf("SEED: 0x%"PRIx64"\n\n", t)
```

So it works in general with a decent C++ compiler and I'd like to keep LTM in that state if possible.

Reply:
I don't think the c++ variant with the static variable is a good idea. You could also write a different macro which is not an expression.

MP_GET_ENDIAN(endian) do...while etc

Reply:
> You could also write a different macro which is not an expression

Of course I could.
Shoudl I?
I think I should, yes.